api/v1/verticadb_webhook.go

```diff
@@ -1021,6 +1025,108 @@ func (v *VerticaDB) hasValidProbeOverride(allErrs field.ErrorList, fieldPath *fi
 	return allErrs
 }
 
+// validateSandboxes validates if provided sandboxes info is correct
+func (v *VerticaDB) validateSandboxes(allErrs field.ErrorList) field.ErrorList {
+	// if vdb does not have any sandboxes, skip this check
+	if len(v.Spec.Sandboxes) == 0 {
+		return allErrs
+	}
+
+	sandboxes := v.Spec.Sandboxes
+	seenSandbox := make(map[string]any)
+	mainClusterImage := v.Spec.Image
+	path := field.NewPath("spec").Child("sandboxes")
+	for _, sandbox := range sandboxes {
+		// check if we have duplicate sandboxes
+		if _, ok := seenSandbox[sandbox.Name]; ok {
+			err := field.Invalid(path,
+				v.Spec.Sandboxes,
+				fmt.Sprintf("sandbox %s already exists", sandbox.Name))
+			allErrs = append(allErrs, err)
+		}
+		seenSandbox[sandbox.Name] = struct{}{}
+		// check if sandbox image is different than main cluster's before sandbox has been setup
+		if sandbox.Image != mainClusterImage && !v.isSandboxInitialized(sandbox.Name) {
```

Comment:
The `sandbox.Image` can be an empty string. That just means it will inherit the image from vdb.spec.image. So, we should allow that.